homeassistant/components/opower/coordinator.py

```diff
@@ -156,32 +169,100 @@ async def _insert_statistics(self) -> None:
                 cost_sum = cast(float, stats[cost_statistic_id][0]["sum"])
                 consumption_sum = cast(float, stats[consumption_statistic_id][0]["sum"])
                 last_stats_time = stats[consumption_statistic_id][0]["start"]
+                if cost_reads_billing:
```

Comment:
The code below is very similar to the code above. Can you avoid duplicate code by extracting it to a function?